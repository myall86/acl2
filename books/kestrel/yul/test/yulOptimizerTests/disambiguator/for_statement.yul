{
    { let a, b }
    {
        function eq(x, y) -> z {}
        for { let a } eq(a, a) { a := a } {
            let b := a
        }
    }
}
// ====
// dialect: yul
// ----
// step: disambiguator
//
// {
//     { let a, b }
//     {
//         function eq(x, y) -> z
//         { }
//         for { let a_1 } eq(a_1, a_1) { a_1 := a_1 }
//         { let b_2 := a_1 }
//     }
// }
