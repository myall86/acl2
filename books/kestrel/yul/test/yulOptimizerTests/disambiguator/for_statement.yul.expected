
 {
     { let a, b }
     {
         function eq(x, y) -> z
         { }
         for { let a_1 } eq(a_1, a_1) { a_1 := a_1 }
         { let b_2 := a_1 }
     }
 }
